File: Java Labs/Patterns On Java/src/com/pattern/iterator/ClothingStore.java
package com.pattern.iterator;


import java.util.ArrayList;
import java.util.List;

public class ClothingStore implements ClothingCollection {
    private List<ClothingItem> clothingItems;

    public ClothingStore() {
        this.clothingItems = new ArrayList<>();
    }

    public void addItem(ClothingItem clothingItem) {
        clothingItems.add(clothingItem);
    }

    @Override
    public Iterator createIterator() {
        return new ClothingIterator(clothingItems);
    }
}

File: Python Labs/com/pattern/iterator/ClothingStore.py
from com.pattern.iterator.ClothingCollection import ClothingCollection
from com.pattern.iterator.ClothingIterator import ClothingIterator


class ClothingStore(ClothingCollection):
    def __init__(self):
        self.clothing_items = []

    def add_item(self, clothing_item):
        self.clothing_items.append(clothing_item)

    def create_iterator(self):
        return ClothingIterator(self.clothing_items)